code/game/turfs/turf.dm

```diff
@@ -538,8 +563,44 @@
 			if(nutri_check.nutriment_factor >0)
 				M.reagents.remove_reagent(R.type, min(R.volume, 10))
 
-//Whatever happens after high temperature fire dies out or thermite reaction works.
-//Should return new turf
+/turf/proc/set_heat_damage_amount(temperature)
+	var/obj/structure/window/W = locate() in range(1, src)
+	if(W)
+		W.adjacent_fire_act(temperature)
+	var/obj/machinery/door/D = locate() in range(1, src)
+	if(D)
+		D.adjacent_fire_act(temperature)
```

Comment:
Shouldn't these be `for` loops instead of `locate()`? Why is only one of each being affected on a 3x3 radius?

Reply:
how the for loop would be made?

Reply:
`for(var/obj/structure/window/W in range(1, src))`